src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -179,6 +193,26 @@ protected Object invokeStep(StepDescriptor d, Object args) {
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
             s = d.newInstance(ps.namedArgs);
+            try {
+                // No point storing empty arguments, and ParallelStep is a special case where we can't store its closure arguments
+                if (ps.namedArgs != null && !(ps.namedArgs.isEmpty()) && isKeepStepInfo() && !(s instanceof ParallelStep)) {
+                    Map<String, Object> actualArgs = ps.namedArgs;
+                    if (d.isMetaStep() && ps.namedArgs.get("delegate") instanceof Map) {
+                        actualArgs = (Map<String,Object>)ps.namedArgs.get("delegate");
```

Comment:
With this in mind, I'll change the implementation here if there's not just a single parameter (as described above) which will handle this case -- though the StepDescriptor of course need to provide its own argumentsToString (it would anyway).